src/Catalog.API/Services/CatalogAI.cs

```diff
@@ -1,38 +1,29 @@
-﻿using Azure;
-using Azure.AI.OpenAI;
+﻿using Azure.AI.OpenAI;
 using Pgvector;
 
 namespace eShop.Catalog.API.Services;
 
 public sealed class CatalogAI : ICatalogAI
 {
-    /// <summary>OpenAI API key for accessing embedding LLM.</summary>
-    private readonly string _aiKey;
-    /// <summary>Optional OpenAI API endpoint.</summary>
-    private readonly string _aiEndpoint;
-    /// <summary>The name of the embedding model to use.</summary>
     private readonly string _aiEmbeddingModel;
+    private readonly OpenAIClient _openAIClient;
 
     /// <summary>The web host environment.</summary>
     private readonly IWebHostEnvironment _environment;
     /// <summary>Logger for use in AI operations.</summary>
     private readonly ILogger _logger;
 
-    public CatalogAI(IOptions<AIOptions> options, IWebHostEnvironment environment, ILogger<CatalogAI> logger)
+    public CatalogAI(IOptions<AIOptions> options, IEnumerable<OpenAIClient> openAIClient, IWebHostEnvironment environment, ILogger<CatalogAI> logger)
```

Comment:
```suggestion
    public CatalogAI(IOptions<AIOptions> options, OpenAIClient openAIClient, IWebHostEnvironment environment, ILogger<CatalogAI> logger)
```

Why is this IEnumerable? I don't think it should be.

Reply:
To check if the service is actually registered.

Reply:
Then make it:

```suggestion
    public CatalogAI(IOptions<AIOptions> options, IWebHostEnvironment environment, ILogger<CatalogAI> logger, OpenAIClient? openAIClient = null)
```